cmd/peer-list/main.go

```diff
@@ -70,6 +72,18 @@ func shellOut(sendStdin, script string) {
 }
 
 func main() {
+	// Custom signal handler for SIGUSR1. This is needed because the
+	// Docker image (percona/percona-xtradb-cluster-operator:$version-haproxy)
+	// sets the STOPSIGNAL to SIGUSR1 to shutdown HAProxy gracefully.
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGUSR1)
+	go func() {
+		for {
+			<-signalChan
+			os.Exit(0)
+		}
```

Comment:
No need in a loop here - the app gonna exit after receiving the signal, so the next step will never happen anyway.
```go
go func() {
	<-signalChan
	os.Exit(0)
}()
```
is enough

Reply:
Other than that LGTM

Reply:
Ah, good to know! I've changed it accordingly. Thanks for letting me know! :)